Dockerfile

```diff
@@ -0,0 +1,18 @@
+
```

Comment:
I am in favour of collecting all the dockerfiles in a single directory. While this is not related to the CI, maybe we can have a folder within .ci that holds all the docker files. Or we can perhaps create a folder in the root directory? What do you think @yunchu

Reply:
I'm afraid a Dockerfile cannot be outside of its root build context with having `COPY . training_extensions/`? I don't think `COPY ..` is possible, but please correct me if I'm wrong

Reply:
Can't we use call the `build` from root and add `-f <path/Dockerfile`

Reply:
I'm not sure that keeps the build context to the repository's root directory, and I also don't see why you'd go through that extra step just to get the Dockerfile out of the root directory

Reply:
Yes, that's why I would like to know what other maintainers think about moving this to another folder. Currently it is not an issue but say in the future you need more configurations like a different ubuntu and cuda version. In that case it would be easier to have multiple docker files as they will have some significant differences. My reference here is the approach used in anomalib (see the first step https://github.com/openvinotoolkit/anomalib/tree/main/.ci#steps). We need containers for ubuntu 18 and 20 running different cuda versions. It keeps the build context while cleanly categorizing multiple docker files. [Lightning](https://github.com/Lightning-AI/lightning/tree/master/dockers) and [Torch](https://github.com/pytorch/pytorch/tree/main/.ci/docker) use similar approach. We can think of a future-proof design but if others are fine with this then I am okay with it as well.

Reply:
Instead of using a Nvidia image, perhaps we can conditionally install CUDA dependencies through the use of build args? It also allows you to use more build args to install different tools conditionally rather than switching between Dockerfiles

Reply:
Sounds good! Let's roll with it. Maybe we can introduce a makefile in the future to make it easier to target different versions.

Reply:
> I am in favour of collecting all the dockerfiles in a single directory. While this is not related to the CI, maybe we can have a folder within .ci that holds all the docker files. Or we can perhaps create a folder in the root directory? What do you think @yunchu

@QSmally @ashwinvaidya17 
sorry for the delayed response. I think it's better to move different subfolder like 'docker' to hold dokerfile instead of using '.ci' because this one is not related with CI and the non-docker user doesn't need to know the context about it.